### PR TITLE
SM-1147: Switch to try_init with pyo3_log

### PR DESCRIPTION
**Pull request by coltonhurst**
## Type of change

- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

When running the `ansible-playbook` command, we needed to export the following variable: `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES`. This suppresses fork safety warnings on Mac OS. With this set, you can safely query for individual secrets. However, if you query for secrets within an Ansible loop or with a Jinja2 template, we receive an error that is caused by setting up a logger more than once. (This error comes from the `pyo3_log` crate.)

If we switch to using the `pyo3_log::try_init` function, we ignore the case where a logger is already set up if the attempt fails. This happens because the `pyo3_log::init` function [can panic](https://github.com/vorner/pyo3-log/blob/70819388bfcb58a7ca91a27179a73cea0abde31e/src/lib.rs#L569). This also introduces panic safety.

## Code changes

- **client.rs:** Switch to `try_init`

## Before you submit

- Please add **unit tests** where it makes sense to do so

### CODE REVIEW

**Review by dani-garcia — APPROVED** [review]
LGTM, we're already doing the same thing for the c and napi bindings, seems like we forgot this one.

https://github.com/bitwarden/sdk/blob/67e743f595e73df9807b6f17f557e327ac0fbc5a/crates/bitwarden-c/src/c.rs#L27

https://github.com/bitwarden/sdk/blob/67e743f595e73df9807b6f17f557e327ac0fbc5a/crates/bitwarden-napi/src/client.rs#L34-L36

**Review by Thomas-Avery — DISMISSED** [review]
LGTM

**Comment by coltonhurst** [comment]
Added a comment and `let` for consistency across crates in [4c5b926](https://github.com/bitwarden/sdk/pull/676/commits/4c5b926f1a59ada1123e58922eb5721dbc979d72).

**Review by dani-garcia — APPROVED** [review]

**Review by tangowithfoxtrot — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Comment by mittkkoo** [comment]
I'm sorry for the dumb question, but how to implement this fix on my side? I did `cargo build`, but what next?

**Comment by jacobfelknor** [comment]
@coltonhurst

Tagging you because you're the PR author and I'm not sure who else to ask, but feel free to delegate. This still appears to be unresolved for users following the official documentation I liked in my above reply for the same reason - no new PyPI package pushed.

 When can the community expect this to be resolved (or how can I help us get there)?

**Comment by coltonhurst** [comment]
@mittkkoo & @jacobfelknor thanks for asking & reaching out, sorry for the delay on this. We have an internal ticket on deck to fix the Python issue, we will be investigating soon!

**Comment by jacobfelknor** [comment]
Great, thank you!

**Comment by betermieux** [comment]
> @mittkkoo & @jacobfelknor thanks for asking & reaching out, sorry for the delay on this. We have an internal ticket on deck to fix the Python issue, we will be investigating soon!

Any news on this? The ansible plugin is not usable if you use more then one secret lookup…

**Comment by felixklauke** [comment]
@coltonhurst Would be great of you could give us an ETA for a new release.

@betermieux In the meantime, I created a dirty little fix for this:

```yaml
- name: Deploy and configure Prometheus
  hosts: service_APP001_Prometheus
  become: true
  pre_tasks:
    - set_fact:
        netbox_api_url: "{{ lookup('bitwarden.secrets.lookup', 'xxx', base_url=base_url )}}"
    - set_fact:
        netbox_api_token: "{{ lookup('bitwarden.secrets.lookup', 'xxx', base_url=base_url )}}"
  roles:
    - prometheus.prometheus.prometheus
```

TL;DR: Put all lookups in it's own (!) set_fact task for now, until we have a release.

**Comment by betermieux** [comment]
> @betermieux In the meantime, I created a dirty little fix for this…

Sorry for not replying earlier, I commented my findings at bitwardens forum:
[Bitwarden lookup plugin for Ansible causes panick errors and “A worker was found in a dead state”](https://community.bitwarden.com/t/bitwarden-lookup-plugin-for-ansible-causes-panick-errors-and-a-worker-was-found-in-a-dead-state/64579)

It basically works if you build a current version of this SDK and replace the packages in your ansible installation. Now someone just needs to publish a current version to the pip repository.
